### PR TITLE
Feature/mongo connection

### PR DESCRIPTION
**Pull request by IILTCC**

### CODE REVIEW

File: MongoConsumer/AppSettings/AppSettingPaths.cs
Comment (eyal-otiker) [inline MongoConsumer/AppSettings/AppSettingPaths.cs:1]: remove this file and put it in startup, like what you did in simulator and telemetry-device

File: MongoConsumer/AppSettings/appsettings.json [hunk 1]
```diff
@@ -9,5 +9,9 @@
       "FiberBoxDownIcd",
       "TelemetryStatistics"
     ]
+  },
+  "MongoConnectionSettings": {
+    "DataBaseName": "telemetryDb",
```
Comment (eyal-otiker) [inline MongoConsumer/AppSettings/appsettings.json:14]: change to TelemtryDB

File: MongoConsumerLibary/MongoConnection/Collections/BaseBoxCollection.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿using System;
+
+namespace MongoConsumerLibary.MongoConnection.Collections
+{
+    public class BaseBoxCollection
```
Comment (eyal-otiker) [inline MongoConsumerLibary/MongoConnection/Collections/BaseBoxCollection.cs:5]: Add two params for this class: Dictionary of all params with decoded params and IcdType

File: MongoConsumerLibary/MongoConnection/Collections/BaseBoxCollection.cs [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+    {
+        public DateTime ExpirationTime { get; set; }
+        public string CompressedData { get; set; }
```
Comment (eyal-otiker) [inline MongoConsumerLibary/MongoConnection/Collections/BaseBoxCollection.cs:8]: remove

File: MongoConsumerLibary/MongoConnection/Collections/FiberBoxDownCollection.cs
Comment (eyal-otiker) [inline MongoConsumerLibary/MongoConnection/Collections/FiberBoxDownCollection.cs:1]: remove

File: MongoConsumerLibary/MongoConnection/Collections/FiberBoxUpCollection.cs
Comment (eyal-otiker) [inline MongoConsumerLibary/MongoConnection/Collections/FiberBoxUpCollection.cs:1]: remove

File: MongoConsumerLibary/MongoConnection/Collections/FlightBoxDownCollection.cs
Comment (eyal-otiker) [inline MongoConsumerLibary/MongoConnection/Collections/FlightBoxDownCollection.cs:1]: remove

File: MongoConsumerLibary/MongoConnection/Collections/FlightBoxUpCollection.cs
Comment (eyal-otiker) [inline MongoConsumerLibary/MongoConnection/Collections/FlightBoxUpCollection.cs:1]: remove

File: MongoConsumerLibary/MongoConnection/Enums/CollectionType.cs
Comment (eyal-otiker) [inline MongoConsumerLibary/MongoConnection/Enums/CollectionType.cs:1]: remove

File: MongoConsumerLibary/MongoConnection/MongoConnection.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        private readonly MongoClient _mongoClient;
+        private readonly IMongoDatabase _database;
+        private readonly Dictionary<CollectionType, IMongoCollection<BaseBoxCollection>> _collections;
+        private readonly MongoSettings _mongoSettings;
+        public MongoConnection(MongoSettings mongoSettings)
+        {
+            _mongoSettings = mongoSettings;
+            _mongoClient = new MongoClient(_mongoSettings.ConnectionUrl);
+            _database = _mongoClient.GetDatabase(_mongoSettings.DataBaseName);
+            _collections = new Dictionary<CollectionType, IMongoCollection<BaseBoxCollection>>();
```
Comment (eyal-otiker) [inline MongoConsumerLibary/MongoConnection/MongoConnection.cs:20]: You don't need collection for every IcdType, put in one collection and add icdType as a property

File: MongoConsumerLibary/MongoConnection/Collections/BaseBoxCollection.cs [hunk 5]
```diff
@@ -0,0 +1,11 @@
+﻿using MongoConsumerLibary.MongoConnection.Enums;
+using System.Collections.Generic;
+
+namespace MongoConsumerLibary.MongoConnection.Collections
+{
+    public class BaseBoxCollection : ExpirableCollection
+    {
+        public BaseBoxType IcdType { get; set; }
+        public Dictionary<string,(int,bool)> DecryptedParameters { get; set; }
```
Comment (eyal-otiker) [inline MongoConsumerLibary/MongoConnection/Collections/BaseBoxCollection.cs:9]: why bool?

File: MongoConsumerLibary/MongoConnection/Enums/BaseBoxType.cs [hunk 6]
```diff
@@ -0,0 +1,10 @@
+﻿namespace MongoConsumerLibary.MongoConnection.Enums
+{
+    public enum BaseBoxType
```
Comment (eyal-otiker) [inline MongoConsumerLibary/MongoConnection/Enums/BaseBoxType.cs:3]: change to IcdType

File: MongoConsumerLibary/MongoConnection/Collections/StatisticsCollection.cs [hunk 7]
```diff
@@ -0,0 +1,9 @@
+﻿using System.Collections.Generic;
+
+namespace MongoConsumerLibary.MongoConnection.Collections
+{
+    public class StatisticsCollection : ExpirableCollection
```
Comment (eyal-otiker) [inline MongoConsumerLibary/MongoConnection/Collections/StatisticsCollection.cs:5]: why use in inheritance, put it in one class

File: MongoConsumerLibary/MongoConnection/Collections/BaseBoxCollection.cs (hunk 5 above)
Comment (IILTCC) [reply MongoConsumerLibary/MongoConnection/Collections/BaseBoxCollection.cs:9]: The dictionary coming from Kafka includes a bool for indicating if the parameter is legal or not

Comment (eyal-otiker) [reply MongoConsumerLibary/MongoConnection/Collections/BaseBoxCollection.cs:9]: Ok, put only the legal in the mongoDB and the type of the variable will be Dictionary<string, int>

File: MongoConsumerLibary/MongoConnection/Collections/StatisticsCollection.cs (hunk 7 above)
Comment (IILTCC) [reply MongoConsumerLibary/MongoConnection/Collections/StatisticsCollection.cs:5]: It is needed for the generic class CollectionEntity

**Pull request merged**